File: cs230/cs230-dz07-nikola_tasic_3698/reverse/src/main/java/rs/ac/metropolitan/cs230/service/TagService.java
package rs.ac.metropolitan.cs230.service;

import rs.ac.metropolitan.cs230.entity.Tag;
import java.util.List;

public interface TagService {

	List<Tag> findAll();

	Tag findById(Long idTag);

	Tag save(Tag tag);

	Tag update(Tag tag);

	boolean delete(Tag tag);

	boolean deleteById(Long idTag);
}

File: it255/it255-pz01-nikola_tasic_3698/projects/admin/src/services/tag.service.ts
import { Injectable } from "@angular/core";
import { HttpClient } from "@angular/common/http";
import { environment } from "../environments/environment";
import { Tag } from "../../../common/src/@types/entity/Tag";

@Injectable({
	providedIn: "root",
})
export class TagService {
	private baseUrl = `${environment.baseUrl}/tags`;

	constructor(private http: HttpClient) {
	}

	public getAll() {
		return this.http.get<Tag[]>(`${this.baseUrl}`).toPromise();
	}

	public save(tag: Tag) {
		return this.http.post<Tag>(`${this.baseUrl}`, tag).toPromise();
	}

	public update(tag: Tag) {
		return this.http.put<Tag>(`${this.baseUrl}`, tag).toPromise();
	}

	public delete(id: number | string) {
		return this.http.delete<void>(`${this.baseUrl}/${id}`).toPromise();
	}
}
